test/Microsoft.AspNetCore.Mvc.FunctionalTests/FileResultTests.cs

```diff
@@ -115,6 +212,52 @@ public FileResultTests(MvcTestFixture<FilesWebSite.Startup> fixture)
             Assert.Equal("This is a sample text from a binary array", body);
         }
 
+        [Theory]
+        [InlineData(0, 6, "This is")]
+        [InlineData(29, 40, "binary array")]
+        [InlineData(0, 50, "This is a sample text from a binary array")]
+        public async Task FileFromBinaryData_ReturnsFile_RangeRequest(long start, long end, string expectedBody)
+        {
+            // Arrange & Act
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://localhost/DownloadFiles/DownloadFromBinaryData");
+            httpRequestMessage.Headers.Range = new RangeHeaderValue(start, end);
+
+            var response = await Client.SendAsync(httpRequestMessage);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.PartialContent, response.StatusCode);
+
+            Assert.NotNull(response.Content.Headers.ContentType);
+            Assert.Equal("text/plain", response.Content.Headers.ContentType.ToString());
+
+            var body = await response.Content.ReadAsStringAsync();
+            Assert.NotNull(body);
+            Assert.Equal(expectedBody, body);
+        }
+
+        [Theory]
+        [InlineData("0-6")]
+        [InlineData("bytes = 11-6")]
+        [InlineData("bytes = 1-4, 5-11")]
+        public async Task FileFromBinaryData_ReturnsFile_RangeRequestNotSatisfiable(string rangeString)
```

Comment:
I could have missed it, but where is the test where a client gets AcceptRanges header in the response even though they did not make a range request? (Like this is how a client can know that ranges are supported)